### DISCUSSION TITLE
Patch2Self introduces direction bias

### DISCUSSION THREAD

**Discussion by m13slash9** [dipy/dipy#2598] [Preprocessing]
Hi, everyone.
I'm looking to denoise some extremity DTI data (12 directions, 6 averages, b = 0 and 650) with patch2self, and the denoised images look really well.
![noisy-denoised](https://user-images.githubusercontent.com/93515611/170689939-6ea8881f-b2d8-477a-8187-052a93c59f23.PNG)

Unfortunately, after the reconstruction the denoised data shows a predominant diffusion direction in the whole volume.
Below are the sample slices reconstructed from the original and denoised datasets, showing the diversity of directions is lost after patch2self. 

Any suggestions, what might be the issue?

Reconstruction of the original data:
![recon_orig](https://user-images.githubusercontent.com/93515611/170690037-d2a6b143-d988-4217-902c-df879bb203ae.PNG)

Reconstruction of the data after patch2self (as the only additional step between original data and reconstruction):
![recon_afterp2s](https://user-images.githubusercontent.com/93515611/170690151-3d363fe2-3e04-488d-a959-235f4ae56275.PNG)

**Comment by ShreyasFadnavis** [comment]
Hi @m13slash9 ! This is quite interesting 👍🏽 


This may solve the problem.
Can you set these parameters for patch2self --> 
`denoised_arr = patch2self(data, bvals, shift_intensity=False, clip_negative_vals=True, patch_radius[1, 1, 1]')`

Can you tell me a little about the dimensionality of the data? I think changing the regressor within Patch2Self may also help.

Thanks,
Shreyas

**Reply by m13slash9** [reply]
Hi, @ShreyasFadnavis thanks for the suggestion, I'll try that.

Also, now when I see your suggestion, I see other options I haven't used (i.e., `clip_negative_vals`), as by now I've been using
`patch2self(diff_data, gradient_table.bvals, model='ols', shift_intensity=True, clip_negative_vals=False)`

The data has the dimensions `(76, 148, 28, 78)` ordered as height, width, slices, diffusion weightings.

I'll post the updates, when the data is processed.

**Reply by m13slash9** [reply]
Indeed,
`patch2self(diff_data, gradient_table.bvals, model='ols', shift_intensity=False, clip_negative_vals=True,patch_radius = [1, 1, 1])`
worked without altering the direction globally.
Though it's quite slower.
![patch111](https://user-images.githubusercontent.com/93515611/170796491-3be4f92a-c318-4adf-af2f-a27f9f6242a6.PNG)

**Reply by ShreyasFadnavis** [reply]
Yes! To make it faster, you can try changing model='ridge'. May I ask what kind of data this is?

One more thing to try is making the patch_radius=[1,1,0]. I'm not sure if this will alter the quality of the output, but worth a shot.

Thanks,
Shreyas

**Reply by m13slash9** [reply]
It's a DTI scan of the forearm. That green line in the middle should probably be the median nerve, though I haven't checked if it really is - the slice was selected kind of randomly for demonstration.

**Reply by ShreyasFadnavis** [reply]
That is very cool! Just for a sanity check, can you also try doing: 
`patch2self(diff_data, gradient_table.bvals, model='ols', shift_intensity=True, clip_negative_vals=False, patch_radius = [1, 1, 1])`

I wanted to check if `shift_intensity` parameter was causing the bias...
